sys/arm64/arm64/unwind.c

```diff
@@ -41,19 +41,18 @@ unwind_frame(struct thread *td, struct unwind_state *frame)
 
 	fp = frame->fp;
 
-	if (!__is_aligned(fp, sizeof(fp)) ||
-	    !kstack_contains(td, fp, sizeof(fp) * 2))
-		return (false);
-
 #ifdef __CHERI_PURE_CAPABILITY__
-	if ((ptraddr_t)fp < cheri_getbase(fp) ||
-	    (ptraddr_t)(fp + sizeof(fp) * 2) > cheri_gettop(fp) ||
-	    cheri_gettag(fp) == 0 ||
-	    (cheri_getperm(fp) & (CHERI_PERM_LOAD | CHERI_PERM_LOAD_CAP)) !=
-	    (CHERI_PERM_LOAD | CHERI_PERM_LOAD_CAP))
+	if (!cheri_is_unsealed_with_perms((void *)fp,
```

Comment:
You could consider adding a wrapper for this check that is something like:

```
bool cheri_can_access(cap, perms, base, length);
```
?